odml/tools/version_converter.py

```diff
@@ -29,69 +30,274 @@ class VersionConverter(object):
+                        val.text = str(value[element])
+                    else:
+                        elem = ET.Element(element)
+                        elem.text = str(value[element])
+                        val.append(elem)
+
+            parent_element.append(val)
+
+    def _convert(self, tree):
+        """
+        Converts an lxml.ElementTree containing a v1.0 odML document to odML v1.1.
+        Unites multiple value objects and moves all supported Value elements to
+        its parent Property. Exports only Document, Section and Property elements,
+        that are supported by odML v1.1.
+        """
+        # Reset status messages
+        self.conversion_log = []
+
+        tree = self._replace_same_name_entities(tree)
         root = tree.getroot()
         root.set("version", FORMAT_VERSION)
 
-        rem_property = []
+        # Handle Values, exclude unsupported Property attributes and unnamed Properties.
+        self._handle_properties(root)
+
+        # Exclude unsupported Section attributes, ignore comments, handle repositories.
+        for sec in root.iter("section"):
+            sec_name = sec.find("name").text
+            for e in sec:
+                if e.tag not in format.Section.arguments_keys and isinstance(e.tag, str):
+                    self._log("[Info] Omitted non-Section attribute "
+                              "'%s: %s/%s'" % (sec_name, e.tag, e.text))
+                    sec.remove(e)
+                    continue
+
+                if e.tag == "repository":
+                    self._handle_repository(e)
+                elif e.tag == "include":
+                    self._handle_include(e)
+
+        # Exclude unsupported Document attributes, ignore comments, handle repositories.
+        for e in root:
+            if e.tag not in format.Document.arguments_keys and isinstance(e.tag, str):
+                self._log("[Info] Omitted non-Document "
+                          "attribute '%s/%s'" % (e.tag, e.text))
+                root.remove(e)
+                continue
+
+            if e.tag == "repository":
+                self._handle_repository(e)
+
+        return tree
+
+    def _handle_include(self, element):
+        """
+        _handle_include checks whether a provided include element is
+        v1.1 compatible and logs a warning message otherwise.
+        :param element: lxml element containing the provided include link.
+        """
+        content = element.text
+        cache = {}
+        term_handler = Terminologies(cache)
+        term = term_handler.load(element.text)
+
+        # If the include url can be loaded and parsed, everything is fine.
+        if term is not None:
+            return
+
+        # If the include url is a v1.0 odml-terminology one,
+        # check whether a v1.1 is available and use it instead.
+        if os.path.join(REPOSITORY_BASE, "v1.0") in element.text:
+            element.text = element.text.replace('v1.0', 'v1.1')
```

Comment:
this will not work on windows, right? os.path.join will use the backslash?!

Reply:
ah, good catch!